lib/livebook_web/live/session_live.ex

```diff
@@ -1223,6 +1267,25 @@ defmodule LivebookWeb.SessionLive do
     {:noreply, socket}
   end
 
+  def handle_info({:set_secret, secret}, socket) do
+    idx = Enum.find_index(socket.assigns.livebook_secrets, &(&1.name == secret.name))
+
+    livebook_secrets =
+      if idx do
+        put_in(socket.assigns.livebook_secrets, [Access.at(idx), :value], secret.value)
+      else
+        socket.assigns.livebook_secrets ++ [secret]
+      end
+      |> Enum.sort()
```

Comment:
To be clear, I meant a map like this:

```suggestion
    livebook_secrets = put_in(socket.assigns.livebook_secrets[secret.name], secret.value)
```

Then you sort when rendering. :)